analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CollectionsShouldImplementGenericInterface.cs

```diff
@@ -17,24 +17,25 @@ class TestClass_05 : IEnumerable, ICollection { } // Noncompliant {{Refactor thi
     // Noncompliant@-1 {{Refactor this collection to implement 'System.Collections.Generic.ICollection<T>'.}}
 
     class TestClass_06 : IEnumerable, ICollection<string> { }
-    class TestClass_07 : IEnumerable, IList<string> { }
-    class TestClass_08 : IEnumerable, IEnumerable<string> { }
-    class TestClass_09 : Collection<string>, IEnumerable { }
-    class TestClass_10<T> : IEnumerable, IList<T> { }
+    class TestClass_07 : ICollection<string>, IEnumerable { }
```

Comment:
I would actually name it something like `GenericFirst` and the previous `GenericLast` to explain a little what it tests. It would also remove the need to rename the numbers on the rest :)